apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+  case status(Node, Pools) of
+    {pending, Ref, _Id} ->
+      NewPools = enable(Node, Pools),
+      {next_state, pools, State#state{pools = NewPools}};
+    _ ->
+      {next_state, zero, State}
+  end;
+
+zero_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  NewPools = disable(Node, Pools),
+  case status(Node, Pools) of
+    {pending, _Ref, Id} ->
+      case task(Category, Id) of
+        undefined ->
+          {next_state, zero, State#state{pools = NewPools}};
+        Task ->
+          rpc:cast(Node, fyler_pool, cancel_task, [Id]),
+          {next_state, tasks, State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}}
+      end;
+    _ ->
+      {next_state, zero, State#state{pools = NewPools}}
+  end;
+
+zero_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools,
+  passive_pools = PassivePools} = State) ->
+
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  NewState = State#state{pools = delete(Node, Pools), passive_pools = [Node|PassivePools]},
+  case fyler_queue:is_empty(NewReTasks) of
+    true ->
+      {next_state, zero, NewState};
+    false ->
+      {next_state, tasks, NewState#state{retasks = NewReTasks}}
+  end;
+
+zero_({cancel_task, Id}, #state{category = Category} = State) ->
+  case ets:lookup(Category, Id) of
+    [#current_task{pool = Node, status = progress}] ->
+      ets:delete(Category, Id),
+      rpc:cast(Node, fyler_pool, cancel_task, [Id]);
+    _ ->
+      ok
+  end,
+  {next_state, zero, State}.
+
+pools_(start, #state{pools = [_]} = State) ->
+  {next_state, pools, State};
+
+pools_(start, #state{} = State) ->
+  {next_state, pools, send_check(State)};
+
+pools_(check, #state{pools = [_]} = State) ->
+  {next_state, pools, State};
+
+pools_(check, #state{pools = [{Pool, _}|_], node_to_id = NodeToId} = State) ->
+  stop_pool(Pool, NodeToId),
+  {next_state, pools, send_check(State)};
+
+pools_({pool_connected, Node}, #state{pools = [Pool], passive_pools = PassivePools} = State) ->
+  NewState = State#state{pools = connect(Node, [Pool]), passive_pools = lists:delete(Node, PassivePools)},
+  {next_state, pools, send_check(NewState)};
+
+pools_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->
+  {next_state, pools, State#state{pools = connect(Node, Pools), passive_pools = lists:delete(Node, PassivePools)}};
+
+pools_({pool_enabled, Node}, #state{pools = Pools} = State) ->
+  {next_state, pools, State#state{pools = enable(Node, Pools)}};
+
+pools_({task_accepted, Ref, Node}, #state{pools = Pools} = State) ->
+  NewPools =
+    case status(Node, Pools) of
+      {pending, Ref, _Id} ->
+        enable(Node, Pools);
+      _ ->
+        Pools
+    end,
+  {next_state, pools, State#state{pools = NewPools}};
+
+pools_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
```

Comment:
Line 286 is too long: pools_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->.